src/app/clients/ClientsPage.tsx

```diff
@@ -34,7 +37,24 @@ const ClientsPage: React.FC<{}> = () => {
     const [tableData, setTableData] = useState<ClientsTableRow[]>([]);
     const [modalIsOpen, setModalIsOpen] = useState<boolean>(false);
     const [activeData, setActiveData] = useState<ExpandedClientDetails | null>(null);
+    const [clientParcelData, setClientParcelData] = useState<ClientParcelTableRow[] | null>(null);
     const theme = useTheme();
+    const router = useRouter();
+
+    const searchParams = useSearchParams();
+    const clientId = searchParams.get("clientId");
+
+    useEffect(() => {
+        if (clientId) {
+            getExpandedClientDetails(clientId).then((data) => {
+                setActiveData(data);
+                setModalIsOpen(true);
+                getExpandedClientParcelsDetails(clientId).then((data) => {
+                    setClientParcelData(data);
+                });
+            });
```

Comment:
It feels like the Parcels page does things much better, which is
- when a parcel is clicked, set the parcel id and immediately open the modal
- pass the parcel id to the modal, which is going to fetch the data and show it to the user

Can we implement something similar on the Clients page too? Currently I think the clients page itself is trying to fetch the data before opening the modal which feels a bit less responsive.